browse/services/document/metadata.py

```diff
@@ -568,12 +577,19 @@ def _parse_metadata_fields(fields: Dict, key_value_block: str) -> None:
         return fields
 
 
+
+@wraps(AbsMetaSession.get_show_email_hash)
+def get_show_email_hash(identifier: Identifier) -> str:
```

Comment:
Could the email hash be only at the UI level? It doesn't really have anything to do with the metadata. The whole email hash might be just a UI feature.

Reply:
I just pushed some changes that use the UI (filter) based approach. Does this seems like a reasonable interim solution for building the /show-email URL?

Reply:
Looks good, can you remove the remaining email hash stuff in metadata.py? It looks like it doesn't need SHOW_EMAIL_SECRET any more.